pyclam/manifold.py

```diff
@@ -133,13 +135,15 @@ def argsamples(self) -> Vector:
             else:
                 n = int(np.sqrt(len(self)))
                 indices = list(np.random.choice(self.argpoints, n, replace=False))
+                indices = [int(i) for i in indices]
 
             # Handle Duplicates.
-            if pdist(self.manifold.data[indices], self.metric).max(initial=0.) == 0.:
+            if self.distance(x1=indices, x2=indices).max(initial=0.) == 0.:
```

Comment:
Recommendation to not use variable names here, this function *almost* justifies being position only. It doesn't matter which comes first.